talk/basicconcepts/headersinterfaces.tex

```diff
@@ -73,3 +73,20 @@
     \end{cppcode*}
   \end{block}
 \end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Header / source separation}
+  \begin{goodpractice}[Header / source]{Headers and source files}
+    \begin{itemize}
+      \item Headers should contain declarations of functions / classes
+      \item Might be included/compiled many times
+      \item Good to keep them short
+      \item Minimise \cppinline{#include} statements
+      \item Put long code in implementation files. Exceptions:
+        \begin{itemize}
+          \item Short functions
+          \item Templates and constexpr functions
+        \end{itemize}
```

Comment:
Maybe add a line that headers are not mandatory and should not be created just for the sake of it. It's only for interfaces that are used somewhere else.